cmd/deploy/endpoints.go

```diff
@@ -93,32 +94,20 @@ func Endpoints(ctx context.Context, k8sLogger *io.K8sLogger) *cobra.Command {
 					return err
 				}
 				options.ManifestPath = model.GetManifestPathFromWorkdir(options.ManifestPath, workdir)
-			}
-
-			ctxResource, err := utils.LoadManifestContext(options.ManifestPath)
-			if err != nil {
-				if oktetoErrors.IsNotExist(err) {
-					ctxResource = &model.ContextResource{}
+				// check whether the manifest file provided by -f exists or not
+				if _, err := fs.Stat(options.ManifestPath); err != nil {
+					return oktetoErrors.UserError{
+						E:    fmt.Errorf("the okteto manifest file '%s' does not exist", options.ManifestPath),
+						Hint: "Check the path to the okteto manifest file",
+					}
 				}
 			}
 
-			if err := ctxResource.UpdateNamespace(options.Namespace); err != nil {
-				return err
-			}
-
-			if err := ctxResource.UpdateContext(options.K8sContext); err != nil {
-				return err
-			}
-
-			ctxOptions := &contextCMD.Options{
-				Context:   ctxResource.Context,
-				Namespace: ctxResource.Namespace,
-			}
-			if options.Output == "" {
-				ctxOptions.Show = true
-			}
-			if err := contextCMD.NewContextCommand().Run(ctx, ctxOptions); err != nil {
-				return err
+			// Loads, updates and uses the context from path. If not found, it creates and uses a new context
+			if err := contextCMD.LoadContextFromPath(ctx, options.Namespace, options.K8sContext, options.ManifestPath, contextCMD.Options{Show: options.Output == ""}); err != nil {
```

Comment:
the `okteto endpoints` command can be used as follows:

```
okteto endpoints --output=json | jq XYZ
```

or

```
okteto endpoints --output=md > README.md
```

In those scenarios we cannot output the plaintext header or it would break those scenarios. For `build` and `deploy` we don't have such use-cases, or at least I am not aware or them, so the `Show` value - only for `endpoints` depend on its `--output` flag.

Worth mentioning that for "plaintext" in most commands of the CLI we use empty string. So by default an empty `--output` flag means "plaintext". So the check is a bit odd. I hope that with 3.0 we can add a third allowed value for `--output plaintext` or `--output text` to make this more explicit.

Reply:
For `deploy` and `destroy` we have the global flag `--log-output` which could be also `json` (I was confusing the flag output with that global flag). I was testing deploy and destroy, and when the `--log-output` flag is set, we are not displaying the context information even if we set `Show: true` in the context.

I see it working fine but I think this is confusing for the user. If I execute `okteto endpoints --log-output=json` it doesn't print anything in the output